pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -279,11 +292,43 @@ func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error
 			Reason:  remoteFinishedCond.Reason,
 			Message: remoteFinishedCond.Message,
 		})
-		return a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
+		return reconcile.Result{}, a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
 	}
 
 	hasReserving, reservingRemote := group.FirstReserving()
 
+	// if the reserving workload is out of sync, remove it
+	if hasReserving {
+		outOfSync := group.local == nil || !equality.Semantic.DeepEqual(group.local.Spec, group.remotes[reservingRemote].Spec)
+		if outOfSync {
+			if err := client.IgnoreNotFound(group.RemoveRemoteObjects(ctx, reservingRemote)); err != nil {
+				log.V(2).Error(err, "Deleting out of sync remote objects", "remote", reservingRemote)
+				return reconcile.Result{}, err
+			}
+			group.remotes[reservingRemote] = nil
+
+			// check if another one exists
+			hasReserving, reservingRemote = group.FirstReserving()
```

Comment:
Should we not repeat the outOfSync check for the new pair: `hasReserving, reservingRemote `?

Reply:
previously , L308 we remove the out of sync one, we check if another also has reservation.

Reply:
Reworked